### PR TITLE
[14.0][ADD] account_invoice_inter_company_queued

### PR DESCRIPTION
**Pull request by ilyasProgrammer**
Perform counter part invoice generation, implemented in module "Inter Company Invoices", using queued jobs.

### CODE REVIEW

**Review by elvise — APPROVED** [review]
Functional review: Perfect!

**Review by francesco-ooops — APPROVED** [review]
functional review ok

**Comment by elvise** [comment]
@pedrobaeza also this one looks ready :)

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
You can't put such code in the main module. Please follow other approach like the one I used in https://github.com/OCA/bank-payment/tree/8.0/account_payment_transfer_reconcile_batch

**Comment by elvise** [comment]
@ilyasProgrammer Can you follow the comment ?

**Comment by ilyasProgrammer** [comment]
@pedrobaeza is it better now? I'm not sure if this exact change that was requested.

File: account_invoice_inter_company/models/account_move.py [hunk 1]
```diff
@@ -34,6 +34,11 @@ def _find_company_from_invoice_partner(self):
     def _post(self, soft=True):
         """Validated invoice generate cross invoice base on company rules"""
         res = super()._post(soft=soft)
+        if not self.env.context.get("queued"):
```
Comment (pedrobaeza) [inline account_invoice_inter_company/models/account_move.py]: Use a more specific context key for avoiding conflicts and do the check here:

```suggestion
        if not self.env.context.get("account_invoice_inter_company_queued"):
```

File: account_invoice_inter_company/views/res_config_settings_view.xml [hunk 2]
```diff
@@ -51,6 +51,20 @@
                         </div>
                     </div>
                 </div>
+                    <div class="col-12 col-lg-6 o_setting_box" id="invoice_queued">
```
Comment (pedrobaeza) [inline account_invoice_inter_company/views/res_config_settings_view.xml]: Don't add this here, but on the new module. Name the div or group you need for hooking it correctly though.

File: account_invoice_inter_company/models/res_config_settings.py [hunk 3]
```diff
@@ -30,6 +30,10 @@ class ResConfigSettings(models.TransientModel):
         "(product where company is defined). Product not related to a "
         "company are visible for all companies.",
     )
+    module_account_invoice_inter_company_queued = fields.Boolean(
```
Comment (pedrobaeza) [inline account_invoice_inter_company/models/res_config_settings.py]: Please don't do that. When migrating modules, this option won't be available until the other module is migrated, and confuse users.

File: account_invoice_inter_company_queued/views/res_config_settings_view.xml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        <field name="model">res.config.settings</field>
+        <field
+            name="inherit_id"
+            ref="account_invoice_inter_company.res_config_settings_view_form"
+        />
+        <field name="arch" type="xml">
+            <xpath expr="//div[@id='account_invoice_inter_company']" position="after">
+                <div class="col-12 col-lg-6 o_setting_box" id="invoice_queued">
+                    <div class="o_setting_left_pane">
+                        <field name="module_account_invoice_inter_company_queued" />
```
Comment (pedrobaeza) [inline account_invoice_inter_company_queued/views/res_config_settings_view.xml:13]: This is like a self-destruction mechanism, hehe

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**